src/Raven.Server/Https/HttpsConnectionMiddleware.cs

```diff
@@ -1,68 +1,43 @@
 using System.Threading.Tasks;
+using Microsoft.AspNetCore.Connections;
+using Microsoft.AspNetCore.Connections.Features;
 using Microsoft.AspNetCore.Http.Features;
-using Microsoft.AspNetCore.Server.Kestrel.Core.Adapter.Internal;
+using Microsoft.AspNetCore.Http.Features.Authentication;
+using Microsoft.AspNetCore.Server.Kestrel.Core;
 using Microsoft.AspNetCore.Server.Kestrel.Https;
-using Sparrow.Logging;
 
 namespace Raven.Server.Https
 {
-
-    public class HttpsConnectionAdapter : IConnectionAdapter
+    public class HttpsConnectionMiddleware
     {
-        private readonly HttpsConnectionAdapterOptions _options;
+        private readonly RavenServer _server;
+        private readonly CipherSuitesPolicy _cipherSuitesPolicy;
 
         // See http://oid-info.com/get/1.3.6.1.5.5.7.3.1
         // Indicates that a certificate can be used as a SSL server certificate
         private const string ServerAuthenticationOid = "1.3.6.1.5.5.7.3.1";
 
-        private static readonly ClosedAdaptedConnection _closedAdaptedConnection = new ClosedAdaptedConnection();
-
         private X509Certificate2 _serverCertificate;
-        private readonly Logger _logger;
 
-        public HttpsConnectionAdapter(X509Certificate2 certificate) : this()
+        public HttpsConnectionMiddleware(RavenServer server, KestrelServerOptions options)
         {
-            SetCertificate(certificate);
-        }
+            _server = server;
+            _cipherSuitesPolicy = server.Configuration.Security.TlsCipherSuites != null && server.Configuration.Security.TlsCipherSuites.Length > 0
+                ? new CipherSuitesPolicy(server.Configuration.Security.TlsCipherSuites)
+                : null;
 
-        public HttpsConnectionAdapter()
-        {
-            _logger = LoggingSource.Instance.GetLogger<HttpsConnectionAdapter>("Server");
-            _options = new HttpsConnectionAdapterOptions();
-        }
-
-
-        public void SetCertificate(X509Certificate2 serverCertificate)
-        {
-            EnsureCertificateIsAllowedForServerAuth(serverCertificate);
-
-            Interlocked.Exchange(ref _serverCertificate, serverCertificate);
-        }
-
-        public bool IsHttps => true;
-
-        public Task<IAdaptedConnection> OnConnectionAsync(ConnectionAdapterContext context)
-        {
-            // Don't trust SslStream not to block.
-            return Task.Run(() => InnerOnConnectionAsync(context));
-        }
-
-        private async Task<IAdaptedConnection> InnerOnConnectionAsync(ConnectionAdapterContext context)
-        {
-            var sslStream = new SslStream(context.ConnectionStream,
-                leaveInnerStreamOpen: false,
-                userCertificateValidationCallback: (sender, certificate, chain, sslPolicyErrors) =>
+            options.ConfigureHttpsDefaults(o =>
+            {
+                o.SslProtocols = SslProtocols.Tls13 | SslProtocols.Tls12;
+                o.CheckCertificateRevocation = true;
```

Comment:
Set to false.
Check that this works in offline mode

Reply:
Need to check `HandshakeTimeout`

Reply:
@iftahbe is checking that